cmd/kubeadm/app/constants/constants.go

```diff
@@ -208,8 +208,32 @@ var (
 
 	// MinimumKubeletVersion specifies the minimum version of kubelet which kubeadm supports
 	MinimumKubeletVersion = version.MustParseSemantic("v1.8.0")
+
+	// SupportedEtcdVersion lists officially supported etcd versions with corresponding kubernetes releases
+	SupportedEtcdVersion = map[uint8]string{
+		8: "3.0.17",
+		9: "3.1.10",
+	}
 )
 
+// EtcdSupportedVersion returns officially supported version of etcd for a specific kubernetes release
+// if passed version is not listed, the function returns nil and an error
+func EtcdSupportedVersion(versionString string) (*version.Version, error) {
```

Comment:
please require the caller to pass in a `*version.Version` here instead of parsing things yourself inline here
I don't want this func to return an error

Reply:
please add an unit test for this :)

Reply:
If somebody passes k8s version which is not on the list?

Reply:
Another thing that bug me, on one hand we tried to hide details from the user of this function (removing converting to uint8 in the previous revision of this pr) which make sense, but here you suggest to force the user to convert to version.Version? I am trying to understand the logic here.

Reply:
ok, I'm not feeling very strongly here, so as you added the unit test I'm fine with this :)